SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp

```diff
@@ -100,31 +101,43 @@ bool USpatialGDKEditorSettings::IsAssemblyNameValid(const FString& Name)
 {
 	const FRegexPattern AssemblyPatternRegex(SpatialConstants::AssemblyPattern);
 	FRegexMatcher RegMatcher(AssemblyPatternRegex, Name);
-
-	return RegMatcher.FindNext();
+	bool valid = RegMatcher.FindNext();
+	if (!valid) {
+		UE_LOG(LogSpatialEditorSettings, Error, TEXT("Assembly name is invalid. It should match the regex: %s"), *SpatialConstants::AssemblyPattern);
```

Comment:
Actually, these errors should be in the call site and not in this method. It's not an error for the method to return a false result of the check, it's an error that in the call site we require a pass to continue.

Reply:
I suggest making this change for all the methods